localstack-core/localstack/services/apigateway/next_gen/execute_api/gateway_response.py

```diff
@@ -1,125 +1,237 @@
-    status_code = 403
+class WafFilteredError(BaseGatewayException):
     type = GatewayResponseType.WAF_FILTERED
+
+
+# File to merge with next_gen/gateway_response when we switch over to it.
+# This will allow in the meantime to expend on functionality without requiring to import from the provider to /next_gen
+from enum import Enum
+
+from localstack.aws.api.apigateway import (
+    GatewayResponse,
+    GatewayResponseType,
+    MapOfStringToString,
+    StatusCode,
+)
+
+
+class GatewayResponseCode(StatusCode, Enum):
+    REQUEST_TOO_LARGE = "413"
+    RESOURCE_NOT_FOUND = "404"
+    AUTHORIZER_CONFIGURATION_ERROR = "500"
+    MISSING_AUTHENTICATION_TOKEN = "403"
+    BAD_REQUEST_BODY = "400"
+    INVALID_SIGNATURE = "403"
+    INVALID_API_KEY = "403"
+    BAD_REQUEST_PARAMETERS = "400"
+    AUTHORIZER_FAILURE = "500"
+    UNAUTHORIZED = "401"
+    INTEGRATION_TIMEOUT = "504"
+    ACCESS_DENIED = "403"
+    DEFAULT_4XX = ""
+    DEFAULT_5XX = ""
+    WAF_FILTERED = "403"
+    QUOTA_EXCEEDED = "429"
+    THROTTLED = "429"
+    API_CONFIGURATION_ERROR = "500"
+    UNSUPPORTED_MEDIA_TYPE = "415"
+    INTEGRATION_FAILURE = "504"
+    EXPIRED_TOKEN = "403"
+
+
+def build_gateway_response(
+    response_type: GatewayResponseType,
+    status_code: StatusCode = None,
+    response_parameters: MapOfStringToString = None,
+    response_templates: MapOfStringToString = None,
+    default_response: bool = True,
+) -> GatewayResponse:
+    """Building a Gateway Response. Non provided attributes will use default."""
+    response = GatewayResponse(
+        responseParameters=response_parameters or {},
+        responseTemplates=response_templates
+        or {"application/json": '{"message":$context.error.messageString}'},
+        responseType=response_type,
+        defaultResponse=default_response,
+    )
+    if status_code or (status_code := GatewayResponseCode[response_type]):
+        # DEFAULT_4XX and DEFAULT_5XX do not have `statusCode` present in the response
+        response["statusCode"] = status_code
+    return response
+
+
+def get_gateway_response_or_default(
+    response_type: GatewayResponseType,
+    gateway_responses: dict[GatewayResponseType, GatewayResponse],
+) -> GatewayResponse:
+    """Utility function that will look for a matching Gateway Response in the following order.
+    - If provided in the gateway_response, return the dicts value
+    - If the DEFAULT_XXX was configured will create a new response
+    - Otherwise we return from DEFAULT_GATEWAY_RESPONSE"""
+
+    if response := gateway_responses.get(response_type):
+        # User configured response
+        return response
+    response_code = GatewayResponseCode[response_type]
+    if response_code == "":
+        # DEFAULT_XXX response do not have a default code
+        return DEFAULT_GATEWAY_RESPONSES.get(response_type)
+    if response_code >= "500":
+        # 5XX response will either get a user configured DEFAULT_5XX or the DEFAULT_GATEWAY_RESPONSES
```

Comment:
Ahh, so there is no default, that makes sense, right! It's much clearer now! thanks!